pkg/build/webhook/controller.go

```diff
@@ -74,7 +78,12 @@ func (c *controller) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	if !proceed {
 		return
 	}
-	build := util.GenerateBuildFromConfig(buildCfg, revision, nil)
+	imageRepos, err := c.imageRepoLister.List(labels.Everything(), labels.Everything())
```

Comment:
my assumption is that the imageRepoLister is going to be access restricted when it is instantiated in master.  How would you propose narrowing it?  The buildConfig could reference any imagerepo in the system, theoretically.